src/libs/SearchUIUtils.ts

```diff
@@ -1776,17 +1795,32 @@ function getSortedTaskData(data: TaskListItemType[], localeCompare: LocaleContex
  * @private
  * Sorts report sections based on a specified column and sort order.
  */
-function getSortedReportData(data: TransactionReportGroupListItemType[], localeCompare: LocaleContextProps['localeCompare']) {
+function getSortedReportData(data: TransactionReportGroupListItemType[], localeCompare: LocaleContextProps['localeCompare'], sortBy?: SearchColumnType, sortOrder?: SortOrder) {
     for (const report of data) {
         report.transactions = getSortedTransactionData(report.transactions, localeCompare, CONST.SEARCH.TABLE_COLUMNS.DATE, CONST.SEARCH.SORT_ORDER.DESC);
     }
 
+    if (!sortBy || !sortOrder) {
+        return data.sort((a, b) => {
+            if (!a.created || !b.created) {
+                return 0;
+            }
+
+            return localeCompare(b.created.toLowerCase(), a.created.toLowerCase());
+        });
+    }
+
+    const sortingProperty = reportColumnNamesToSortingProperty[sortBy as keyof typeof reportColumnNamesToSortingProperty];
+
+    if (!sortingProperty) {
+        return data;
+    }
+
     return data.sort((a, b) => {
-        if (!a.created || !b.created) {
-            return 0;
-        }
+        const aValue = a[sortingProperty as keyof TransactionReportGroupListItemType];
+        const bValue = b[sortingProperty as keyof TransactionReportGroupListItemType];
 
-        return localeCompare(b.created.toLowerCase(), a.created.toLowerCase());
+        return compareValues(aValue, bValue, sortOrder, sortingProperty, localeCompare);
```

Comment:
We need to update the logic when sorting by total, since IOU reports always have positive amounts, whereas expense reports are usually negative amounts. This is the logic we have in the BE:

```
reportType === 'iou' ? report.total : -report.total
```

Reply:
Same with the status. It seems like we're sorting based on the state/statusNum, but we should sort alphabetically

Reply:
Another thing. When sorting by `total` we're sending `amount` to the BE. We should update that to `total`